service/tablet_allocator.cc

```diff
@@ -1009,12 +1009,46 @@ class load_balancer {
                     // FIXME: we can try another pair of non-colocated replicas of same sibling tablets.
                     lblogger.debug("Load limit reached, unable to emit migration for replica ({}, {}) to co-habit the replica ({}, {})",
                         t2_id, src, t1_id, dst);
-                    return;
+                    return false;
                 }
                 apply_load(nodes, mig_streaming_info);
 
+                utils::get_local_injector().inject("disallow_cross_rack_colocation", [&] {
```

Comment:
Not sure if this shouldn't be checked unconditionally, outside of error injection, when `rack_awareness` is true.

Reply:
or at least under `#ifndef SCYLLA_BUILD_MODE_RELEASE`

Reply:
You're right it's a good thing to check outside of error injection path with rack awareness.